game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProMoveUtils.java

```diff
@@ -185,10 +187,12 @@ public static void calculateAmphibRoutes(
       for (final Unit transport : amphibAttackMap.keySet()) {
         int movesLeft = TripleAUnit.get(transport).getMovementLeft().intValue();
         Territory transportTerritory = ProData.unitTerritoryMap.get(transport);
+        moves.newSequence();
 
         // Check if units are already loaded or not
-        final List<Unit> loadedUnits = new ArrayList<>();
-        final List<Unit> remainingUnitsToLoad = new ArrayList<>();
+        final var loadedUnits = new ArrayList<Unit>();
```

Comment:
I believe its needed. You need to specify `Unit` either on the left or right side. Otherwise it would initialize to an ArrayList of objects.

Reply:
Sure enough, I was thinking you'd get a 'type cannot be inferred'

Reply:
I'm fairly sure it will let you do it though I would think IDEs/checkstyle/etc might give you a warning. Its one of the reasons I don't really like `var` in a lot cases as its kind of dangerous if you don't know what you are doing.

Reply:
I'm curious how have you seen it being dangerous? Type cannot be inferred is a compiler error, it's for when the non-generic type (IE: List or Map) cannot be inferred. For generics, defaulting up to an object is likely going to also run into an error later unless you are actually pulling objects from the collections. On another note, this usage I think is okay, the scope of the variable is too large anyways, where the variable is used, you can't read the declaration anyways, so it does not matter either way.